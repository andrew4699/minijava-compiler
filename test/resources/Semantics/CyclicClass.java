class Factorial{
	public static void main(String[] a){
		System.out.println(new A().ComputeFac(10));
	}
}

class A extends B {
	int a;
	boolean b;

	public int ComputeFac(int num){
		int num_aux ;
		if (num < 1)
				num_aux = 1 ;
		else 
				num_aux = num * (this.ComputeFac(num-1)) ;
		return num_aux ;
	}
}

class B extends A {
	int a;
	boolean b;

	public int ComputeFac(int num){
		int num_aux ;
		if (num < 1)
				num_aux = 1 ;
		else 
				num_aux = num * (this.ComputeFac(num-1)) ;
		return num_aux ;
	}
}
