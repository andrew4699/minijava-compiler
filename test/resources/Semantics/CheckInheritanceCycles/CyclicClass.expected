[GlobalSymbolTable]: Class A is part of an inheritance cycle
[GlobalSymbolTable]: Class B is part of an inheritance cycle
